client/go/outline/client.go

```diff
@@ -33,11 +33,30 @@ type Client struct {
 	transport.PacketListener
 }
 
+// NewClientResult represents the result of [NewClientAndReturnError].
+type NewClientResult struct {
+	Client *Client
+	Error  *platerrors.PlatformError
+}
+
+// NewClientAndReturnError creates a new Outline client from a configuration string.
+//
+// TODO: rename this function to NewClient once all platforms have switched from the old NewClient
+func NewClientAndReturnError(transportConfig string) *NewClientResult {
+	client, err := NewClient(transportConfig)
+	return &NewClientResult{
+		Client: client,
+		Error:  platerrors.ToPlatformError(err),
+	}
+}
+
 // NewClient creates a new Outline client from a configuration string.
+//
+// Deprecated: Use [NewClientAndReturnError] instead.
 func NewClient(transportConfig string) (*Client, error) {
```

Comment:
Is this still being used? If not, please delete it. Same for the other deprecated code.

We no longer need to be backwards-compatible in our Go interface with the app, since we build them atomically.

Reply:
Oh, I guess it's still used on Apple?

Reply:
Yes.